src/client/sandbox/event/message.js

```diff
@@ -169,12 +169,16 @@ export default class MessageSandbox extends SandboxBase {
         if (!this._isIFrameRemoved() && (isIFrameWithoutSrc || !isCrossDomainWindows(targetWindow, this.window) &&
                                                                targetWindow[this.RECEIVE_MSG_FN])) {
             //NOTE: postMessage delay imitation
-            nativeMethods.setTimeout.call(this.topWindow, () =>
+            nativeMethods.setTimeout.call(this.topWindow, () => {
+                // NOTE: GH-171
+                if (!isSafari || this.window.top) {
```

Comment:
This is cryptic. It took a while for me to decifer, what we shouldn't send message if we missing `window.top` in Safari. I suggest the following:
```js
// NOTE: add comment here about situation when it may happen with GH-issue reference
var isDetachedWindow = isSafari && !this.window.top;

if (!detachedWindow)
```
BTW, should we check for Safari here? We haven't encountered it in other browsers, but wouldn't it be better to use feature detection instead of browser check here (in case it will appear in other browser)?